2007-08-08  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* bfd/opncls.c (bfd_openr_iovec): Fix OPEN parameter macro expansion.

--- ./bfd/opncls.c	24 Jul 2007 19:58:06 -0000	1.48
+++ ./bfd/opncls.c	8 Aug 2007 19:12:15 -0000
@@ -545,7 +545,8 @@ bfd_openr_iovec (const char *filename, c
   nbfd->filename = filename;
   nbfd->direction = read_direction;
 
-  stream = open (nbfd, open_closure);
+  /* `open (...)' would get expanded by an the open(2) syscall macro.  */
+  stream = (*open) (nbfd, open_closure);
   if (stream == NULL)
     {
       _bfd_delete_bfd (nbfd);



2007-08-08  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* serial.c (serial_open): Fix OPEN parameter macro expansion.

--- ./gdb/serial.c	8 Apr 2007 15:20:07 -0000	1.29
+++ ./gdb/serial.c	8 Aug 2007 20:32:10 -0000
@@ -213,7 +213,8 @@ serial_open (const char *name)
   scb->bufp = scb->buf;
   scb->error_fd = -1;
 
-  if (scb->ops->open (scb, open_name))
+  /* `...->open (...)' would get expanded by an the open(2) syscall macro.  */
+  if ((*scb->ops->open) (scb, open_name))
     {
       xfree (scb);
       return NULL;
